homeassistant/components/group/manifest.json

```diff
@@ -5,8 +5,10 @@
     "alarm_control_panel",
     "climate",
     "device_tracker",
+    "integration",
     "person",
     "plant",
+    "utility_meter",
```

Comment:
I don't think this is a solution, but working around the actual issue.

As in, groups shouldn't be awaiting integrations. For example, if there is a custom integration that have this issue, they can't add to this list either.

I think this needs to be fixed in integration & utility meter to maybe store/restore these specific values earlier.

../Frenck

Reply:
Agree with that 👍 
Also possibly we should do a retry on calculating device class, uom etc. if none of the source entities are available at startup but it doesn't change the fact it needs to be considered in integration and utility meter to not be swapping these attributes.

Reply:
I see, hanging on my nails on the core cliff already, I'll have another try. Any details were to look appreciated.

Reply:
So what changed in 2024.2.0 that caused this issue to occur?

Reply:
Group sensor is only calculating device class, state class and UoM once during startup so if the source entities are not providing the correct input then this calculation is then obviously also not correct.
So it's important that any input entities are providing the correct attributes directly as group sensor loads.

If this isn't the case then one can also specify these on the group sensor and then there is no calculation needed (as the user has defined them himself).

Reply:
@frenck, Are we mixing things?

> As in, groups shouldn't be awaiting integrations. For example, if there is a custom integration that have this issue, they can't add to this list either.

Integration is [Riemann sum Integral](https://www.home-assistant.io/integrations/integration/) , not integration or custom integrations.
```json
  "domain": "integration",
  "name": "Integral",
  "after_dependencies": ["counter"],
  "codeowners": ["@dgomes"],
  "config_flow": true,
  "documentation": "https://www.home-assistant.io/integrations/integration",
  "integration_type": "helper",
  "iot_class": "local_push",
  "quality_scale": "internal"
```
The group is summing a bunch of these Riemann sum integrals, hence the dependency I would think. 

> So it's important that any input entities are providing the correct attributes directly as group sensor loads.

This is working
```txt
device integration sensor +
device integration sensor +-> Group summing 
device integration sensor +
```

The failing groups are kind of second level,  
```txt
device integration sensor -> rieman sum integral +
device integration sensor -> rieman sum integral +-> Group summing 
device integration sensor -> rieman sum integral +
```
or even
```txt
device integration sensor -> rieman sum integral -> daily utiliy meter +
device integration sensor -> rieman sum integral -> daily utiliy meter +-> Group summing 
device integration sensor -> rieman sum integral -> daily utiliy meter +
```

Reply:
> So it's important that any input entities are providing the correct attributes directly as group sensor loads.
>
> If this isn't the case then one can also specify these on the group sensor and then there is no calculation needed 

Checking the entity registry, the `device integration sensor`s all have uom W in the registry and their group sum is working. (first example above)

Riemann Integral Sum entities of these have no uom in the entity registry. Riemann Integral calculates the uom based on the setting of the integral and uom of the source entity.The uom of the source entity (W) is available in the entity registry, but calc needs to run. Is like this for long time. Uom can not be set using the UI or YAML.

The utility meter on the Riemann Integral sum has no uom in entity registry, it uses the uom of the source entity. That is kWh from the Riemann Integral, which need to be running and have calculated the uom. Uom can not be set using UI or YAML.

Group itself also has no uom in entity registry, it determines it from its sources as well. Uom can not be set using UI or YAML.

All 3 seem to share same behavior. In my case setting the after_dependency solved the issue. As one can also use utility_Meter or Integral sum with a group as source that might need other dependency. 

So making sure these 3 store uom in entity registry might solve the issue I guess?

Not sure how this applies for templates, as @ISNL-PimP is reporting issues with these. Templates have an after_dependency on group already.

Reply:
I have made https://github.com/home-assistant/core/pull/119021 which would recalculate the group sensor attributes of device class, state class and uom if not available during startup which at least would partially solve the problem for entities information not available during group sensor startup.

Haven't looked into either integration nor utility meter but I would have guessed both of them was writing to entity registry the correct attributes so it seems as perhaps there is something not working in them as it's supposed too.

Reply:
> I have made https://github.com/home-assistant/core/pull/119021 which would recalculate the group

Thanks @gjohansson-ST, I've tested it with my configuration and it resolves the issue for the Group sums on `utility_meter` and `integration`. 

! Nested groups (Group sum of other Group Sum) still have the issue if one of the source groups get loaded after the group summing it. !

Overall #119021 is definitively an improvement, but as group sum of group sum is a realistic scenario it isn't there yet.

Looking at alternatives to try, I also build the same sums using the `combine state of several sensors` helper (looks like that's min-max integration), as one would use them these days. These don't show the issue reported. Haven't found a simple way for migrating from `group sum` to `combine sum` without state and/or history loss if possible at all. With a clear recipe for this, if feasible, I would migrate to avoid future complications.

 >Haven't looked into either integration nor utility meter but I would have guessed both of them was writing to entity registry the correct attributes so it seems as perhaps there is something not working in them as it's supposed too.

These have uom `null` in the core.entity_registry and calculate it when setup using the source entity and options from core.config_entries or configuration.yaml. Similar to group itself it seems as these have null too.